test_scripts/API/VehicleData/StabilityControlsStatus/016_SubscribeVehicleData_StabilityControlsStatus_invalid_hmi_response_and_other_parameters.lua

```diff
@@ -0,0 +1,103 @@
+---------------------------------------------------------------------------------------------------
+-- Proposal: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0253-New-vehicle-data-StabilityControlsStatus.md
+--
+-- Description: Check subscription on StabilityControlsStatus data via SubscribeVehicleData RPC
+--   with invalid HMI response and other parameters
+--
+-- Precondition:
+-- 1) SDL and HMI are started
+-- 2) App is registered
+-- 3) PTU is successfully performed
+-- 4) App is activated
+
+-- Steps:
+-- 1) App sends SubscribeVehicleData request with (stabilityControlsStatus = true,
+--   gps = true) to SDL
+-- SDL sends VehicleInfo.SubscribeVehicleData request with (stabilityControlsStatus = true,
+--   gps = true) to HMI
+-- HMI sends invalid VehicleInfo.SubscribeVehicleData response related to stabilityControlsStatus
+-- SDL sends SubscribeVehicleData response with (success: false, resultCode: "GENERIC_ERROR")
+-- 2) HMI sends VehicleInfo.OnVehicleData notifications with StabilityControlsStatus or GPS data
+-- SDL does not send OnVehicleData notification with received from HMI data to App
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local common = require('test_scripts/API/VehicleData/commonVehicleData')
+
+--[[ Local variables ]]
+local vdNameSuccess = "gps"
+local vdNameInvalidRes = "stabilityControlsStatus"
+local mobileRequest = {
+  [vdNameSuccess] = true,
+  [vdNameInvalidRes] = true
+}
+local hmiRequest = mobileRequest
+
+local hmiResponseEmpty = {
+  [vdNameInvalidRes] = {},
+  [vdNameSuccess] = common.getResponse("SUCCESS", vdNameSuccess)
+}
+local hmiResponseInvalidValues = {
+  [vdNameInvalidRes] = { resultCode = "SUCCESS", dataType = "invalidType" },
+  [vdNameSuccess] = common.getResponse("SUCCESS", vdNameSuccess)
+}
+local hmiResponseAbsent = {
+  [vdNameSuccess] = common.getResponse("SUCCESS", vdNameSuccess)
+}
+local hmiResponseIncorrectType = {
+  [vdNameInvalidRes] = { resultCode = "SUCCESS" },
+  [vdNameSuccess] = common.getResponse("SUCCESS", vdNameSuccess)
+}
+local hmiResponseNotAStructure = {
```

Comment:
@mvorobio As a suggestion - you can put all these structures in one table and go through it in a loop, as was done here https://github.com/LuxoftSDL/sdl_atf_test_scripts/pull/16/files#diff-82ed425198b71780df166654274cbccb

Reply:
@LitvinenkoIra I thought about it but lack of unified reaction stopped me from doing this )